// kclsu-components/dist/kclsu-components/image-slider-auto.entry.js
import { r as registerInstance, h, g as getElement } from './index-65ac344c.js';

const imageSliderAutoCss = ".slider-container{position:relative;height:100%;width:100%;overflow:hidden}.slider-container img{width:100%;height:100%;object-fit:cover}.hide{display:none;}.slide{object-fit:cover;}.show{display:block;-webkit-animation:fade-in 1.2s cubic-bezier(0.390, 0.575, 0.565, 1.000) both;animation:fade-in 1.2s cubic-bezier(0.390, 0.575, 0.565, 1.000) both}.slide-in-right{-webkit-animation:slide-in-right 1s cubic-bezier(0.250, 0.460, 0.450, 0.940) both;animation:slide-in-right 1s cubic-bezier(0.250, 0.460, 0.450, 0.940) both}.slide-out-left{-webkit-animation:slide-out-left 1s cubic-bezier(0.550, 0.085, 0.680, 0.530) both;animation:slide-out-left 1s cubic-bezier(0.550, 0.085, 0.680, 0.530) both}.fade-in{-webkit-animation:fade-in 1.2s cubic-bezier(0.390, 0.575, 0.565, 1.000) both;animation:fade-in 1.2s cubic-bezier(0.390, 0.575, 0.565, 1.000) both}@-webkit-keyframes slide-in-right{0%{-webkit-transform:translateX(1000px);transform:translateX(1000px);opacity:0}100%{-webkit-transform:translateX(0);transform:translateX(0);opacity:1}}@keyframes slide-in-right{0%{-webkit-transform:translateX(1000px);transform:translateX(1000px);opacity:0}100%{-webkit-transform:translateX(0);transform:translateX(0);opacity:1}}@-webkit-keyframes slide-out-left{0%{-webkit-transform:translateX(0);transform:translateX(0);opacity:1}100%{-webkit-transform:translateX(-1000px);transform:translateX(-1000px);opacity:0}}@keyframes slide-out-left{0%{-webkit-transform:translateX(0);transform:translateX(0);opacity:1}100%{-webkit-transform:translateX(-1000px);transform:translateX(-1000px);opacity:0}}@-webkit-keyframes fade-in{0%{opacity:0}100%{opacity:1}}@keyframes fade-in{0%{opacity:0}100%{opacity:1}}";

const ImageSliderAuto = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
        this.classes = 'slide-in-right';
        this.slideIndex = 0;
        this.slideI = 0;
    }
    componentDidLoad() {
        this.oldWay();
        this.timer = setInterval(() => this.oldWay(), this.interval);
    }
    oldWay() {
        let i;
        let slides;
        slides = this.host.querySelectorAll('.slide');
        for (i = 0; i < slides.length; i++) {
            if (!slides[i].classList.contains('hide'))
                slides[i].classList.add('hide');
            if (slides[i].classList.contains('show')) {
                slides[i].classList.remove('show');
            }
        }
        this.slideI++;
        if (this.slideI > slides.length) {
            this.slideI = 1;
        }
        slides[this.slideI - 1].classList.remove('hide');
        slides[this.slideI - 1].classList.add('show');
    }
    changeImage(images) {
        let i = this.slideIndex;
        if (i >= images.length - 1) {
            i = 0;
        }
        else
            i = i + 1;
        this.slideIndex = i;
    }
    render() {
        return (h("div", { class: "slider-container" }, this.images.map(img => h("img", { class: "slide", src: img, alt: "" }))));
    }
    get host() { return getElement(this); }
};
ImageSliderAuto.style = imageSliderAutoCss;

export { ImageSliderAuto as image_slider_auto };
