// kclsu-components/dist/kclsu-components/lazy-image.entry.js
import { r as registerInstance, h } from './index-65ac344c.js';

const lazyImageCss = ".loadingimage{width:100%;height:100%;overflow:hidden}.hide{opacity:0}.show{opacity:1}.fit{width:100%;height:100%;object-fit:cover}@-webkit-keyframes scale-in-center{0%{-webkit-transform:scale(0);transform:scale(0);opacity:1}100%{-webkit-transform:scale(1);transform:scale(1);opacity:1}}@keyframes scale-in-center{0%{-webkit-transform:scale(0);transform:scale(0);opacity:1}100%{-webkit-transform:scale(1);transform:scale(1);opacity:1}}";

const LazyImage = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
        this.image = 'https://res.cloudinary.com/kclsu-media/image/upload/f_auto,fl_any_format,g_center,q_100/v1581516201/website_uploads/KCLSU%20Brand/Bzcl1r6L_400x400_se7grm.jpg';
        this.imageloading = true;
        this.imageclasses = ['fit', 'hide'];
    }
    imageLoaded() {
        this.imageloading = false;
        this.imageclasses = ['fit', 'show', 'scale-in-center'];
    }
    render() {
        return (h("div", { class: "loadingimage" }, h("loading-spinner", { show: this.imageloading }), h("img", { onLoad: () => this.imageLoaded(), class: this.imageclasses.join(' '), src: this.image })));
    }
};
LazyImage.style = lazyImageCss;

export { LazyImage as lazy_image };
