// kclsu-components/dist/kclsu-components/tab-content.entry.js
import { r as registerInstance, h } from './index-65ac344c.js';

const tabContentCss = ".tab-content{display:none;background:#f1f1ee;padding:40px 30px}.tab-content-active{display:block}@media screen and (max-width: 780px){.tab-content{padding:1em}}";

const TabContent = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
        this.active = false;
        this.isSelected = this.active;
    }
    render() {
        const classes = {
            'tab-content': true,
            'tab-content-active': this.isSelected
        };
        if (this.isSelected) {
            return (h("section", { class: classes, role: "tabpanel", "aria-labelledby": this.name }, h("slot", null)));
        }
        else {
            return (h("section", { class: classes, tabindex: "-1", hidden: true, role: "tabpanel", "aria-labelledby": this.name }, h("slot", null)));
        }
    }
    onActiveChanged(newValue) {
        this.isSelected = newValue;
    }
    select() {
        this.isSelected = true;
    }
    unselect() {
        this.isSelected = false;
    }
    static get watchers() { return {
        "active": ["onActiveChanged"]
    }; }
};
TabContent.style = tabContentCss;

export { TabContent as tab_content };
